handlers/handlers.go

```diff
@@ -17,25 +18,28 @@ import (
 	"github.com/xeipuuv/gojsonschema"
 )
 
-type StreamInfo struct {
-	callbackUrl string
-}
-
 type CatalystAPIHandlersCollection struct {
 	MistClient  MistAPIClient
-	StreamCache map[string]StreamInfo
+	StreamCache *StreamCache
 }
 
 func (d *CatalystAPIHandlersCollection) Ok() httprouter.Handle {
-	return func(w http.ResponseWriter, req *http.Request, _ httprouter.Params) {
+	return func(w http.ResponseWriter, _ *http.Request, _ httprouter.Params) {
 		io.WriteString(w, "OK")
 	}
 }
 
-func (d *CatalystAPIHandlersCollection) TranscodeSegment() httprouter.Handle {
+// Takes mpegts segment as input, ingests in Mist under SOURCE_PREFIX streamName, transcodes into RENDITION_PREFIX stream then pushes renditions to specified destination.
+// Manually starts MistLivepeerProc binary. For this we require port of B-node that listens on 127.0.0.1 .
+// When `MistProcLivepeer` starts, SOURCE_PREFIX stream is taken as input and (will automatically - WIP) start SOURCE_PREFIX stream.
+// No trigger is registered on SOURCE_PREFIX so configured sourceUrl is used.
+// Also `MistProcLivepeer` outputs to RENDITION_PREFIX stream where we listen for triggers to start new push to rendition destination.
```

Comment:
I'm missing the context, but why is `MistLivepeeerProc` a separate binary? What is its purpose? I think you could add this explanation here. Or even better, move the code related to starting this binary to a separate function and add this explanation there.